crates/aptos-logger/src/aptos_logger.rs

```diff
@@ -314,10 +330,28 @@ impl AptosDataBuilder {
 
                 filter_builder.build()
             };
+            let telemetry_filter = {
+                let mut filter_builder = Filter::builder();
+
+                if self.is_async && self.remote_log_tx.is_some() {
+                    if env::var(RUST_LOG_TELEMETRY).is_ok() {
+                        filter_builder.with_env(RUST_LOG_TELEMETRY);
+                    } else if env::var(RUST_LOG).is_ok() {
+                        filter_builder.with_env(RUST_LOG);
+                    } else {
+                        filter_builder.filter_level(self.telemetry_level.into());
```

Comment:
I think we should move this condition above `RUST_LOG` check. Do we want to emit debug logs if RUST_LOG=debug? I think no.

Reply:
Good catch! I will remove `RUST_LOG` in this entirely. I think that will make more sense